### PR TITLE
Contributing and overall doc comments

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR addresses some issues mentioned in #1305 
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Moves CONTRIBUTING to the top-level
* Adds licensing information to the README
* Adds `sphinx_autodoc_typehints` to the docs
* Removes `Mapping` types from call signatures (`dict` was more appropriate)
* Refactors `HISTORY.rst` into `CHANGES.rst` and `Changleog`

### Does this PR introduce a breaking change?

Yes. The docs have been completely reorganized.

### Other information:

### CODE REVIEW

**Review by huard — APPROVED** [review]

**Comment by Zeitsperre** [comment]
It looks as though `sphinx-autodoc-typehints` might not work for us (130+ warnings on ReadTheDocs) due to a bug that occurs when statically parsing the type information for custom classes in call signatures: https://github.com/tox-dev/sphinx-autodoc-typehints/issues/186

Will check for a workaround or a way of squelching the warnings.

**Review by tlogan2000 — APPROVED** [review]
LGTM

**Comment by Zeitsperre** [comment]
I've restructured the API docs to collect all the relevant submodule APIs (with the exceptions of i18n and data quality checks). I think this works much better. There are other comments that could be addressed (`usage` could flow much), but I think this is fine for a start.

I'll wait on more reviews before merging.

**Comment by Zeitsperre** [comment]
## Update

@Ouranosinc/xclim-core 

I've staged a page for us to explain how xclim is different from other projects (see final comment in #1305) with some ipsum dolor text and links to relevant projects. I can get this started, but would appreciate a hand.

It also made sense to concentrate most of the usage targets under the `API` page, except for the climate indices. The reason is that the `API` page is already enormous. This also makes it easier to navigate the climate indices, as they now live on their own page.

All these changes were causing hyperlink issues, so to resolve them I decided on the following:
- The `modules` section of the docs contains hyperlink targets for all functions except for `indicators` and `indices`
- The hyperlink targets for `indicators` and `indices` will point to entries in `User API` or under the `Indices` pages.

This makes it so that we can finally have valid targets that we can share with users that no longer shows the `_hidden` path, e.g.:

Instead of:
 * https://xclim.readthedocs.io/en/stable/xclim.indicators.atmos.html#xclim.indicators.atmos._temperature.biologically_effective_degree_days

We now have: 
 * (proposed) https://xclim.readthedocs.io/en/stable/api.html#xclim.indicators.atmos.biologically_effective_degree_days

IMO, this is a major improvement, but it's a bit of a departure, so I'd like to hear peoples' thoughts on this.

Thanks!

**Review by bzah — APPROVED** [review]
My 2 cents:
I really like having all the indices listed in the menu.
I also like that the API is a single page even though it's a bit "slow" to load.

A few suggestions (not specific to the PR but to xclim's doc) :
 - I find "ALL MODULES" > xclim and "USER API" > API sections redundant.
 - In the long run it could be interesting to divide the documentations into categories to group doc pages that belongs together. It could be categories such as the ones proposed in [diataxis](https://diataxis.fr/). It's really tedious to do on existing doc though.
 - I find it unusual to use the word "History" for the Change logs. I would prefer "Changelog" (like dask) or "Release Notes" like (like pandas or numpy).
 - I would suggest putting a sentence + link to `Indicator` on the `Indices` page, to encourage user to use the Indicator class instead.  As per my understanding, you really want indices to be  a technical topic in xclim and Indicators to be the user frontend right ?

Overall: great changes thanks!
_Link to PR's readthedoc build I used for review: https://xclim--1308.org.readthedocs.build/en/1308/readme.html ._

File: docs/explanation.rst
Comment (bzah) [inline docs/explanation.rst:17]:
```suggestion
   - `Their comparison page for icclim and xclim <https://icclim.readthedocs.io/en/stable/explanation/xclim_and_icclim.html>`_
```

**Comment by Zeitsperre** [comment]
> My 2 cents: I really like having all the indices listed in the menu. I also like that the API is a single page even though it's a bit "slow" to load.
> 
> A few suggestions (not specific to the PR but to xclim's doc) :
> 
>     * I find "ALL MODULES" > xclim and "USER API" > API sections redundant.
> 
>     * In the long run it could be interesting to divide the documentations into categories to group doc pages that belongs together. It could be categories such as the ones proposed in [diataxis](https://diataxis.fr/). It's really tedious to do on existing doc though.
> 
>     * I find it unusual to use the word "History" for the Change logs. I would prefer "Changelog" (like dask) or "Release Notes" like (like pandas or numpy).
> 
>     * I would suggest putting a sentence + link to `Indicator` on the `Indices` page, to encourage user to use the Indicator class instead.  As per my understanding, you really want indices to be  a technical topic in xclim and Indicators to be the user frontend right ?
> 
> 
> Overall: great changes thanks! _Link to PR's readthedoc build I used for review: https://xclim--1308.org.readthedocs.build/en/1308/readme.html ._

**Comment by Zeitsperre** [comment]
I've staged the `Purpose` page to explain what xclim is and how it differs from other projects. Once we have a more finalized paper, we can pull information directly from that.

I'll be adding one or two comments in the section this afternoon, but aside from that, I think this PR is in good shape and will address several points raised by the reviewer. If there are no other comments, I'll be merging this by the end of the day.

**Review by huard — APPROVED** [review]

**Pull request merged**
